docs/ref/signals.txt

```diff
@@ -99,6 +99,19 @@ Arguments sent with this signal:
 ``instance``
     The actual instance of the model that's just been created.
 
+    .. note::
+
+        ``instance._state`` isn't set before sending the ``post_init`` signal,
+        so ``_state`` attributes have always default values. For example,
+        ``_state.db`` is ``None`` and cannot be used to check an ``instance``
+        database.
+
+.. warning::
+
+    For performance reasons, you shouldn't perform queries in receivers of
+    ``pre_init`` or ``post_init`` signals because they would be executed on
+    every queryset iteration.
```

Comment:
"every queryset iteration" — Do you mean ≈as each instance is created during queryset interation? (Currently might be read to mean once per call to __iter__, i.e. once at the beginning of the loop, which might not be so bad...)

Maybe `because they would be executed for each instance returned during queryset iteration`.

Reply:
Yes, so as often as `post/pre_init()` calls, i.e. N+1 additional queries on every queryset iteration.